pkg/runtime/grpc/instances.go

```diff
@@ -147,17 +147,49 @@ func (r *Runtime) createGadgetInstance(gadgetCtx runtime.GadgetContext, runtimeP
 		EventBufferLength: runtimeParams.Get(ParamEventBufferLength).AsInt32(), // default for now
 	}
 
+	targets, explicitTargets, err := r.getTargets(gadgetCtx.Context(), runtimeParams)
+	if err != nil {
+		return fmt.Errorf("getting targets: %w", err)
+	}
+
+	// if targets have explicitly been listed, add them to the `Nodes` list
+	if explicitTargets {
+		for _, t := range targets {
+			instanceRequest.GadgetInstance.Nodes = append(instanceRequest.GadgetInstance.Nodes, t.node)
+		}
+	}
+
+	var listMutex sync.Mutex
+	var nodeList []string
+	ids := make(map[string][]string)
+	var lastID string
+
 	err = r.runInstanceManagerClientForTargets(gadgetCtx.Context(), runtimeParams, func(target target, client api.GadgetInstanceManagerClient) error {
 		gadgetCtx.Logger().Debugf("creating gadget on node %q", target.node)
 		res, err := client.CreateGadgetInstance(gadgetCtx.Context(), instanceRequest)
 		if err != nil {
 			return fmt.Errorf("creating gadget on node %q: %w", target.node, err)
 		}
-		gadgetCtx.Logger().Infof("installed on node %q as %q", target.node, res.GadgetInstance.Id)
+		listMutex.Lock()
+		nodeList = append(nodeList, target.node)
+		ids[res.GadgetInstance.Id] = append(ids[res.GadgetInstance.Id], target.node)
+		lastID = res.GadgetInstance.Id
+		listMutex.Unlock()
 		return nil
 	})
 	if err != nil {
 		return fmt.Errorf("creating gadget instance: %w", err)
 	}
+
+	if len(ids) > 1 {
+		// this can only happen if the server refused to use the given id (which should not happen with the current
+		// implementations) and we're deploying on multiple targets where each target would choose its own id
+		for k, v := range ids {
+			gadgetCtx.Logger().Infof("installed as %q (nodes %+v)", k, v)
+		}
+		return nil
+	}
```

Comment:
mmm, do we want to give the server the possibility of rejecting an id? I think it'll become difficult to handle if a gadget is installed with different ids on different nodes.

IMO if the server wants to reject an ID, the whole creation should fail instead.

Reply:
I'd leave the decision up to the store implementation (for now this can't happen). On the client side, however, we can either error out (wouldn't be helpful to the user) or just take the info as received and report it to the user - which is the way I have chosen here. I don't want that case to just be unhandled and always assume that there's only one ID.